walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,49 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
 
 public final class FindMeetingQuery {
-  public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
-  }
-}
+    public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
+        Set<String> requestedAttendees = request.getAttendees();
+        long requestedDuration = request.getDuration();
+        
+        if (requestedDuration > TimeRange.WHOLE_DAY.duration()) {
+            return new ArrayList();
+        }
+        
+        if (events.isEmpty()) {
+            return Arrays.asList(TimeRange.WHOLE_DAY);
+        }
+        
+        Collection<TimeRange> queryResponse = findQualifyingTimeRanges(requestedAttendees, events, requestedDuration);
+        
+        return queryResponse; 
+    }
+
+    private Collection<TimeRange> findQualifyingTimeRanges(Set<String> requestedAttendees, Collection<Event> events, long requestedDuration) {
+        Collection<TimeRange> qualifyingTimeRanges = new ArrayList();
+        List<Event> eventsList = new ArrayList(events);
+        Collections.sort(eventsList, Event.ORDER_BY_START);
```

Comment:
You can use stream to collect to Immutable list. Something like:
events.stream()
  .sorted(Comparator)
  .collect(toImmutableList());

Reply:
Done